sdk-core/src/main/java/care/data4life/sdk/migration/RecordCompatibilityService.kt

```diff
@@ -19,138 +19,80 @@ package care.data4life.sdk.migration
 import care.data4life.crypto.GCKey
 import care.data4life.sdk.crypto.CryptoContract
 import care.data4life.sdk.network.NetworkingContract
-import care.data4life.sdk.network.model.NetworkModelContract.EncryptedRecord
+import care.data4life.sdk.network.util.SearchTagsBuilder
 import care.data4life.sdk.tag.Annotations
-import care.data4life.sdk.tag.EncryptedTagsAndAnnotations
-import care.data4life.sdk.tag.TagCryptoHelper
 import care.data4life.sdk.tag.TaggingContract
 import care.data4life.sdk.tag.TaggingContract.Companion.ANNOTATION_KEY
+import care.data4life.sdk.tag.TaggingContract.Companion.DELIMITER
 import care.data4life.sdk.tag.Tags
-import io.reactivex.Observable
-import io.reactivex.Single
-import io.reactivex.functions.BiFunction
-import java.io.IOException
 
+// see: https://gesundheitscloud.atlassian.net/browse/SDK-572
 // see: https://gesundheitscloud.atlassian.net/browse/SDK-525
 @Migration("This class should only be used due to migration purpose.")
 class RecordCompatibilityService internal constructor(
-    private val apiService: NetworkingContract.Service,
-    private val tagCryptoService: TaggingContract.CryptoService,
     private val cryptoService: CryptoContract.Service,
-    private val tagHelper: TaggingContract.Helper = TagCryptoHelper
+    private val tagCryptoService: TaggingContract.CryptoService,
+    private val compatibilityEncoder: MigrationContract.CompatibilityEncoder = CompatibilityEncoder,
+    private val searchTagsBuilderFactory: NetworkingContract.SearchTagsBuilderFactory = SearchTagsBuilder
 ) : MigrationContract.CompatibilityService {
-    private fun encrypt(
-        plainTags: Tags,
-        plainAnnotations: Annotations
-    ): Pair<EncryptedTagsAndAnnotations, EncryptedTagsAndAnnotations> {
-        val tagEncryptionKey = cryptoService.fetchTagEncryptionKey()
-        return Pair(
-            tagCryptoService.encryptTagsAndAnnotations(
-                plainTags,
-                plainAnnotations,
-                tagEncryptionKey
-            ),
-            encryptTags(plainTags, tagEncryptionKey)
-                .also { encryptedTags ->
-                    encryptedTags.addAll(
-                        encryptAnnotations(plainAnnotations, tagEncryptionKey)
-                    )
-                }
+    private fun encryptTags(
+        tagEncryptionKey: GCKey,
+        tagGroupKey: String,
+        tagOrGroup: Triple<String, String, String>
```

Comment:
@alessioborraccino -> is this still an issue?

Reply:
not an issue, maybe would it make sense to make it a key value pair? 
`tagKeyValuePair: (key: String, valueGroup: Triple<String, String, String>) `
although if the Pair is unlabeled maybe it just confuses more

Reply:
You mean  Pair<String, Triple<String, String, String>>   ? As for the Triple - Kotlin has no labeled Pairs out of the box...however, since in this case this can be a simple data class - you like that I go for it?

Reply:
Id always prefer a simple data class over tuples/pairs/triples just for clarifications on whats inside

Reply:
Why not a List? It is anyway converted to one later and so it stays open for later modification

Reply:
You like it more this way?

Reply:
yes much better. thanks